libqtile/confreader.py

```diff
@@ -53,15 +54,21 @@ class Config:
         "wmname",
     ]
 
-    def __init__(self, file_path=None, **settings):
+    def __init__(self, file_path=None, kore=None, **settings):
         """Create a Config() object from settings
 
         Only attributes found in Config.settings_keys will be added to object.
         config attribute precedence is 1.) **settings 2.) self 3.) default_config
         """
+        self.file_path = file_path
+        self.kore = kore
```

Comment:
Having the config reader hold the core object seems a bit heavy handed here. Is this something that can be passed in to `load` and `validate`? That would also give some finer control over which instances the validation is run.

Reply:
Why do you find this heavy handed? I find it boring to pass it again if we already gave it at instanciation. Also, this is what permits to remove the XCore import from libqtile/core/manager.py :)

Reply:
The config object should not have to know what backend core it is using, if it is needed for the sake of validation that is one thing. The import you mention should never have been there in the first place, that should have been using the core that is attached to the manager to do the verification rather than creating an XCore.

Reply:
We can see as we pull out X-specific functionality if it makes sense to leave this like this or if it will help for it to be passed in on verification.